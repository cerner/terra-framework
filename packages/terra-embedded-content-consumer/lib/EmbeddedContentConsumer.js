"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _xfc = require("xfc");
var _excluded = ["src", "onMount", "onLaunch", "onAuthorize", "options", "eventHandlers"];
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var propTypes = {
  /**
   * The source URL of the content to load.
   */
  src: _propTypes.default.string.isRequired,
  /**
   * Notifies the component that the container has been mounted. Provides a reference
   * to this component to allow triggering messages on the embedded application.
   */
  onMount: _propTypes.default.func,
  /**
   * Notifies the component that the container has been launched.
   */
  onLaunch: _propTypes.default.func,
  /**
   * Notifies the component that the container has been authorized.
   */
  onAuthorize: _propTypes.default.func,
  /**
   * The component can be configured with consumer frame options.
   * See xfc consumer configuration for details: https://github.com/cerner/xfc
   */
  options: _propTypes.default.object,
  /**
  * A set of event handlers keyed by the event name.
  * Note: Binding the event handler is necessary to make `this` work in the callback.
  */
  eventHandlers: _propTypes.default.arrayOf(_propTypes.default.shape({
    key: _propTypes.default.string,
    handler: _propTypes.default.func
  }))
};
var EmbeddedContentConsumer = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(EmbeddedContentConsumer, _React$Component);
  var _super = _createSuper(EmbeddedContentConsumer);
  function EmbeddedContentConsumer() {
    (0, _classCallCheck2.default)(this, EmbeddedContentConsumer);
    return _super.apply(this, arguments);
  }
  (0, _createClass2.default)(EmbeddedContentConsumer, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      // Mount the provided source as the application into the content wrapper.
      this.xfcFrame = _xfc.Consumer.mount(this.embeddedContentWrapper, this.props.src, this.props.options);

      // Notify that the consumer frame has mounted.
      if (this.props.onMount) {
        this.props.onMount(this.xfcFrame);
      }

      // Attach the event handlers to the xfc frame.
      this.addEventListener('xfc.launched', this.props.onLaunch);
      this.addEventListener('xfc.authorized', this.props.onAuthorize);

      // Attach custom event handlers to the xfc frame.
      this.addEventListeners(this.props.eventHandlers);
    }
  }, {
    key: "addEventListener",
    value: function addEventListener(eventName, eventHandler) {
      if (eventName && eventHandler) {
        this.xfcFrame.on(eventName, eventHandler);
      }
    }
  }, {
    key: "addEventListeners",
    value: function addEventListeners(customEvents) {
      var _this = this;
      (customEvents || []).forEach(function (e) {
        return _this.addEventListener(e.key, e.handler);
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;
      var _this$props = this.props,
        src = _this$props.src,
        onMount = _this$props.onMount,
        onLaunch = _this$props.onLaunch,
        onAuthorize = _this$props.onAuthorize,
        options = _this$props.options,
        eventHandlers = _this$props.eventHandlers,
        customProps = (0, _objectWithoutProperties2.default)(_this$props, _excluded);
      return /*#__PURE__*/_react.default.createElement("div", (0, _extends2.default)({}, customProps, {
        ref: function ref(element) {
          _this2.embeddedContentWrapper = element;
        }
      }));
    }
  }]);
  return EmbeddedContentConsumer;
}(_react.default.Component);
EmbeddedContentConsumer.propTypes = propTypes;
var _default = EmbeddedContentConsumer;
exports.default = _default;