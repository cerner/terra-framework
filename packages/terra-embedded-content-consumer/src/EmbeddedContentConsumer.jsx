import React from 'react';
import PropTypes from 'prop-types';
import { Consumer } from 'xfc';

const propTypes = {
  /**
   * The source URL of the content to load.
   */
  src: PropTypes.string.isRequired,
  /**
   * Indicates whether the content inside of the iframe should be scrollable or not. Scrolling is default to `true`.
   * When set to `false`, scrolling is disabled.
   * When scrolling is enabled, it is possible to use keyboard navigation to scroll the content even when there is no
   * interactable element inside the content.
   */
  isScrollable: PropTypes.bool,
  /**
   * Notifies the component that the container has been mounted. Provides a reference
   * to this component to allow triggering messages on the embedded application.
   */
  onMount: PropTypes.func,
  /**
   * Notifies the component that the container has been launched.
   */
  onLaunch: PropTypes.func,
  /**
   * Notifies the component that the container has been authorized.
   */
  onAuthorize: PropTypes.func,
  /**
   * The component can be configured with consumer frame options.
   *
   * Example `options` object:
   *
   * ```
   * options = {{
   *   iframeAttrs: {
   *     id: 'iframe-id', // Sets iframe id
   *     srcdoc: 'inline html', // Sets the inline HTML content to render within the iframe
   *     width: '100%', // Width of the iframe
   *     height: '400px', // Height of the iframe, must not be 100%
   *   },
   *   resizeConfig: {
   *     scrolling: true, // true - enable scrolling within the frame, false - disable scrolling within the frame
   *   },
   * }}
   * ```
   *
   * Note: when `srcdoc` is set, this will override the `src` URL.
   * It's important to know that when this option is used,
   * the [xfc](https://github.com/cerner/xfc) library does not manage
   * [auto resizing](https://github.com/cerner/xfc#iframe-resizing-config)
   * of the content of the iframe even if `resizeConfig` option is set. It's
   * important to specify the `width` and `height` of the frame.
   *
   * See xfc consumer configuration for details: https://github.com/cerner/xfc
   *
   */
  options: PropTypes.object,
  /**
  * A set of event handlers keyed by the event name.
  * Note: Binding the event handler is necessary to make `this` work in the callback.
  */
  eventHandlers: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    handler: PropTypes.func,
  })),
};

const defaultProps = {
  isScrollable: true,
};

class EmbeddedContentConsumer extends React.Component {
  componentDidMount() {
    // Merging the iframe options props
    const frameOptions = { ...this.props.options };

    if (frameOptions.iframeAttrs === null
      || frameOptions.iframeAttrs === undefined) {
      frameOptions.iframeAttrs = {};
    }

    if (frameOptions.resizeConfig === null
      || frameOptions.resizeConfig === undefined) {
      frameOptions.resizeConfig = {};
    }

    if (this.props.isScrollable === false) {
      Object.assign(frameOptions.resizeConfig, { scrolling: false });
      Object.assign(frameOptions.iframeAttrs, { scrolling: 'no' });
    } else {
      Object.assign(frameOptions.resizeConfig, { scrolling: true });
      Object.assign(frameOptions.iframeAttrs, { scrolling: 'yes' });
    }

    // Mount the provided source as the application into the content wrapper.
    this.xfcFrame = Consumer.mount(this.embeddedContentWrapper, this.props.src, this.props.options);

    // Notify that the consumer frame has mounted.
    if (this.props.onMount) {
      this.props.onMount(this.xfcFrame);
    }

    // Attach the event handlers to the xfc frame.
    this.addEventListener('xfc.launched', this.props.onLaunch);
    this.addEventListener('xfc.authorized', this.props.onAuthorize);

    // Attach custom event handlers to the xfc frame.
    this.addEventListeners(this.props.eventHandlers);
  }

  addEventListener(eventName, eventHandler) {
    if (eventName && eventHandler) {
      this.xfcFrame.on(eventName, eventHandler);
    }
  }

  addEventListeners(customEvents) {
    (customEvents || []).forEach(e => this.addEventListener(e.key, e.handler));
  }

  render() {
    const {
      src,
      isScrollable,
      onMount,
      onLaunch,
      onAuthorize,
      options,
      eventHandlers,
      ...customProps
    } = this.props;

    return (
      <div
        {...customProps}
        ref={(element) => { this.embeddedContentWrapper = element; }}
      />
    );
  }
}

EmbeddedContentConsumer.propTypes = propTypes;
EmbeddedContentConsumer.defaultProps = defaultProps;

export default EmbeddedContentConsumer;
