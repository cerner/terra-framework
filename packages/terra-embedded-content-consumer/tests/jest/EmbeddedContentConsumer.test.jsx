import React from 'react';
import { Consumer } from 'xfc';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { mountWithIntl } from 'terra-enzyme-intl';
import EmbeddedContentConsumer from '../../lib/EmbeddedContentConsumer';

beforeAll(() => {
  jest.spyOn(console, 'warn').mockImplementation();
});

afterAll(() => {
  console.warn.mockRestore(); // eslint-disable-line no-console
});

// Snapshot Tests
describe(EmbeddedContentConsumer, () => {
  it('should render the embedded content consumer container', () => {
    let frame;
    const onMount = (xfcFrame) => { frame = xfcFrame; };

    const embeddedContentConsumer = (
      <div>
        <EmbeddedContentConsumer
          title="basic provider"
          src="/terra-framework/#/raw/provider/cerner-terra-framework-docs/embedded-content-consumer/providers/basic-provider"
          onMount={onMount}
        />
      </div>
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);
    expect(wrapper).toMatchSnapshot();

    frame.unmount();
  });

  it('should render the embedded content consumer with options attributes', () => {
    let frame;
    const onMount = (xfcFrame) => { frame = xfcFrame; };

    const inlineHtml = '<p><b>Inline HTML Content</b></p><p>This is an inline html content that can be used to render the content into the frame.</p>';
    const frameOptions = {
      iframeAttrs: {
        srcdoc: inlineHtml,
        width: '100%',
        height: '100px',
      },
    };

    const embeddedContentConsumer = (
      <div>
        <EmbeddedContentConsumer
          src=""
          title="inline html content"
          options={frameOptions}
          onMount={onMount}
        />
      </div>
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);
    expect(wrapper).toMatchSnapshot();

    frame.unmount();
  });

  it('should render the embedded content consumer with title prop', () => {
    let frame;
    const onMount = (xfcFrame) => { frame = xfcFrame; };

    const embeddedContentConsumer = (
      <div>
        <EmbeddedContentConsumer
          src="https://www.cdc.gov/"
          title="title for the frame content"
          onMount={onMount}
        />
      </div>
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);
    expect(wrapper).toMatchSnapshot();

    frame.unmount();
  });

  it('should render the embedded content consumer with title prop and options prop', () => {
    let frame;
    const onMount = (xfcFrame) => { frame = xfcFrame; };

    const frameOptions = { iframeAttrs: { title: 'title of content set from options.iframeAttrs' } };
    const embeddedContentConsumer = (
      <div>
        <EmbeddedContentConsumer
          src="https://www.cdc.gov/"
          title="title that will be used for the frame content"
          options={frameOptions}
          onMount={onMount}
        />
      </div>
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);
    expect(wrapper).toMatchSnapshot();

    frame.unmount();
  });

  it('should render the embedded content consumer with custom class names', () => {
    let frame;
    const onMount = (xfcFrame) => { frame = xfcFrame; };

    const embeddedContentConsumer = (
      <div>
        <EmbeddedContentConsumer
          src="https://www.cdc.gov/"
          title="custom class name"
          className="container"
          onMount={onMount}
        />
      </div>
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);
    expect(wrapper).toMatchSnapshot();

    frame.unmount();
  });

  it('should validate the inputs', () => {
    const src = 'https://www.cdc.gov/';
    const title = 'frame content title';
    let frame;
    const onMount = (xfcFrame) => { frame = xfcFrame; };
    const onLaunch = () => { };
    const onAuthorize = () => { };
    const resizeConfig = { scrolling: false, fixedHeight: '100%', fixedWidth: '100%' };
    const options = { secret: 'SecretKey', resizeConfig, iframeAttrs: { title: 'frame content title' } };
    const customEvents = [{ key: 'eventKey', handler: () => { } }];

    const embeddedContentConsumer = (
      <EmbeddedContentConsumer
        src={src}
        title={title}
        onMount={onMount}
        onLaunch={onLaunch}
        onAuthorize={onAuthorize}
        options={options}
        eventHandlers={customEvents}
      />
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);

    expect(frame).toBeTruthy();
    expect(wrapper.instance().props.src).toBe(src);
    expect(wrapper.instance().props.title).toEqual(title);
    expect(frame.iframe.title).toEqual(title);
    expect(wrapper.instance().props.onMount).toBe(onMount);
    expect(wrapper.instance().props.onLaunch).toBe(onLaunch);
    expect(wrapper.instance().props.onAuthorize).toBe(onAuthorize);
    expect(wrapper.instance().props.options).toBe(options);
    expect(wrapper.instance().props.options.resizeConfig).toBe(resizeConfig);
    expect(wrapper.instance().props.eventHandlers).toBe(customEvents);

    frame.unmount();
  });

  it('should render visually hidden text before and after the embedded content consumer container', () => {
    const frameOptions = { iframeAttrs: { title: 'title of content set from options.iframeAttrs' } };
    const embeddedContentConsumer = (
      <EmbeddedContentConsumer
        src="https://www.google.com/"
        options={frameOptions}
      />
    );

    Consumer.init();
    const wrapper = mountWithIntl(embeddedContentConsumer);

    expect(wrapper.find('VisuallyHiddenText').at(0).prop('text')).toEqual('Terra.embeddedContentConsumer.beginEmbeddedContent');
    expect(wrapper.find('VisuallyHiddenText').at(1).prop('text')).toEqual('Terra.embeddedContentConsumer.endEmbeddedContent');
    expect(wrapper).toMatchSnapshot();
  });
});

it('sets appropriate config option when resizeConfig.scrolling is true', () => {
  const embeddedContentConsumer = (
    <div>
      <EmbeddedContentConsumer
        src="https://www.google.com/"
        options={{ resizeConfig: { scrolling: true } }}
      />
    </div>
  );

  const wrapper = shallow(embeddedContentConsumer);
  expect(wrapper).toMatchSnapshot();
});

it('sets appropriate config option when resizeConfig.scrolling is false', () => {
  const embeddedContentConsumer = (
    <div>
      <EmbeddedContentConsumer
        src="https://www.google.com/"
        options={{ resizeConfig: { scrolling: false } }}
      />
    </div>
  );

  const wrapper = shallow(embeddedContentConsumer);
  expect(wrapper).toMatchSnapshot();
});
