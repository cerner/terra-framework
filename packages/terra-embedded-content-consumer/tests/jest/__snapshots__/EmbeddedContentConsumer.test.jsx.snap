// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InjectIntl should render the embedded content consumer container 1`] = `
<div
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(EmbeddedContentConsumer)
    onMount={[Function]}
    src="/terra-framework/#/raw/provider/cerner-terra-framework-docs/embedded-content-consumer/providers/basic-provider"
    title="basic provider"
  >
    <EmbeddedContentConsumer
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onMount={[Function]}
      src="/terra-framework/#/raw/provider/cerner-terra-framework-docs/embedded-content-consumer/providers/basic-provider"
      title="basic provider"
    >
      <div
        className="container"
      >
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.beginEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.beginEmbeddedContent
          </span>
        </VisuallyHiddenText>
        <div
          className="embedded-content-wrapper"
          title="basic provider"
        />
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.endEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.endEmbeddedContent
          </span>
        </VisuallyHiddenText>
      </div>
    </EmbeddedContentConsumer>
  </InjectIntl(EmbeddedContentConsumer)>
</div>
`;

exports[`InjectIntl should render the embedded content consumer with custom class names 1`] = `
<div
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(EmbeddedContentConsumer)
    className="container"
    onMount={[Function]}
    src="https://www.cdc.gov/"
    title="custom class name"
  >
    <EmbeddedContentConsumer
      className="container"
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onMount={[Function]}
      src="https://www.cdc.gov/"
      title="custom class name"
    >
      <div
        className="container"
      >
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.beginEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.beginEmbeddedContent
          </span>
        </VisuallyHiddenText>
        <div
          className="embedded-content-wrapper"
          title="custom class name"
        />
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.endEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.endEmbeddedContent
          </span>
        </VisuallyHiddenText>
      </div>
    </EmbeddedContentConsumer>
  </InjectIntl(EmbeddedContentConsumer)>
</div>
`;

exports[`InjectIntl should render the embedded content consumer with options attributes 1`] = `
<div
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(EmbeddedContentConsumer)
    onMount={[Function]}
    options={
      Object {
        "iframeAttrs": Object {
          "height": "100px",
          "srcdoc": "<p><b>Inline HTML Content</b></p><p>This is an inline html content that can be used to render the content into the frame.</p>",
          "width": "100%",
        },
      }
    }
    src=""
    title="inline html content"
  >
    <EmbeddedContentConsumer
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onMount={[Function]}
      options={
        Object {
          "iframeAttrs": Object {
            "height": "100px",
            "srcdoc": "<p><b>Inline HTML Content</b></p><p>This is an inline html content that can be used to render the content into the frame.</p>",
            "width": "100%",
          },
        }
      }
      src=""
      title="inline html content"
    >
      <div
        className="container"
      >
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.beginEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.beginEmbeddedContent
          </span>
        </VisuallyHiddenText>
        <div
          className="embedded-content-wrapper"
          title="inline html content"
        />
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.endEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.endEmbeddedContent
          </span>
        </VisuallyHiddenText>
      </div>
    </EmbeddedContentConsumer>
  </InjectIntl(EmbeddedContentConsumer)>
</div>
`;

exports[`InjectIntl should render the embedded content consumer with title prop 1`] = `
<div
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(EmbeddedContentConsumer)
    onMount={[Function]}
    src="https://www.cdc.gov/"
    title="title for the frame content"
  >
    <EmbeddedContentConsumer
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onMount={[Function]}
      src="https://www.cdc.gov/"
      title="title for the frame content"
    >
      <div
        className="container"
      >
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.beginEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.beginEmbeddedContent
          </span>
        </VisuallyHiddenText>
        <div
          className="embedded-content-wrapper"
          title="title for the frame content"
        />
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.endEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.endEmbeddedContent
          </span>
        </VisuallyHiddenText>
      </div>
    </EmbeddedContentConsumer>
  </InjectIntl(EmbeddedContentConsumer)>
</div>
`;

exports[`InjectIntl should render the embedded content consumer with title prop and options prop 1`] = `
<div
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <InjectIntl(EmbeddedContentConsumer)
    onMount={[Function]}
    options={
      Object {
        "iframeAttrs": Object {
          "title": "title of content set from options.iframeAttrs",
        },
      }
    }
    src="https://www.cdc.gov/"
    title="title that will be used for the frame content"
  >
    <EmbeddedContentConsumer
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      onMount={[Function]}
      options={
        Object {
          "iframeAttrs": Object {
            "title": "title of content set from options.iframeAttrs",
          },
        }
      }
      src="https://www.cdc.gov/"
      title="title that will be used for the frame content"
    >
      <div
        className="container"
      >
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.beginEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.beginEmbeddedContent
          </span>
        </VisuallyHiddenText>
        <div
          className="embedded-content-wrapper"
          title="title that will be used for the frame content"
        />
        <VisuallyHiddenText
          className="visually-hidden-text"
          text="Terra.embeddedContentConsumer.endEmbeddedContent"
        >
          <span
            className="visually-hidden-text visually-hidden-text"
          >
            Terra.embeddedContentConsumer.endEmbeddedContent
          </span>
        </VisuallyHiddenText>
      </div>
    </EmbeddedContentConsumer>
  </InjectIntl(EmbeddedContentConsumer)>
</div>
`;

exports[`InjectIntl should render visually hidden text before and after the embedded content consumer container 1`] = `
<InjectIntl(EmbeddedContentConsumer)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  options={
    Object {
      "iframeAttrs": Object {
        "title": "title of content set from options.iframeAttrs",
      },
    }
  }
  src="https://www.google.com/"
>
  <EmbeddedContentConsumer
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    options={
      Object {
        "iframeAttrs": Object {
          "title": "title of content set from options.iframeAttrs",
        },
      }
    }
    src="https://www.google.com/"
  >
    <div
      className="container"
    >
      <VisuallyHiddenText
        className="visually-hidden-text"
        text="Terra.embeddedContentConsumer.beginEmbeddedContent"
      >
        <span
          className="visually-hidden-text visually-hidden-text"
        >
          Terra.embeddedContentConsumer.beginEmbeddedContent
        </span>
      </VisuallyHiddenText>
      <div
        className="embedded-content-wrapper"
      />
      <VisuallyHiddenText
        className="visually-hidden-text"
        text="Terra.embeddedContentConsumer.endEmbeddedContent"
      >
        <span
          className="visually-hidden-text visually-hidden-text"
        >
          Terra.embeddedContentConsumer.endEmbeddedContent
        </span>
      </VisuallyHiddenText>
    </div>
  </EmbeddedContentConsumer>
</InjectIntl(EmbeddedContentConsumer)>
`;

exports[`sets appropriate config option when resizeConfig.scrolling is false 1`] = `
<div>
  <InjectIntl(EmbeddedContentConsumer)
    options={
      Object {
        "resizeConfig": Object {
          "scrolling": false,
        },
      }
    }
    src="https://www.google.com/"
  />
</div>
`;

exports[`sets appropriate config option when resizeConfig.scrolling is true 1`] = `
<div>
  <InjectIntl(EmbeddedContentConsumer)
    options={
      Object {
        "resizeConfig": Object {
          "scrolling": true,
        },
      }
    }
    src="https://www.google.com/"
  />
</div>
`;
